lock/src/main/java/org/infinispan/lock/impl/lock/ClusteredLockImpl.java

```diff
@@ -221,34 +255,95 @@ public void entryRemoved(CacheEntryRemovedEvent event) {
       }
    }
 
+   private void retryPendingUnlocks() {
+      RequestHolder nextRequestor = null;
+      log.tracef("LOCK[%s] Unlock pending requests size[%d] in %s", getName(), pendingUnlocks.size(), originator);
+      while (!pendingUnlocks.isEmpty() && !isChangeViewUnlockInProgress()) {
+         UnlockRequestHolder unlockRequestHolder = pendingUnlocks.poll();
+         unlock(unlockRequestHolder);
+      }
+   }
+
+   private void retryPendingRequests() {
+      if (isChangeViewUnlockInProgress()) {
+         log.tracef("LOCK[%s] Hold pending requests while view change unlock is happening in %s", getName(), originator);
```

Comment:
Why is this necessary? Could we have a request from a leaver that is executed during (or even after) the view change unlock?

Reply:
this is what we discussed before, holding requests to wait the unlock over.

Reply:
Ok, I'd just like some comments with the scenario that this is trying to fix.

Reply:
ok